src/interface_alertmanager.py

```diff
@@ -0,0 +1,52 @@
+import json
+import logging
+
+logger = logging.getLogger()
+
+from ops.framework import (
+    EventSource,
+    Object,
+    ObjectEvents,
+)
+from ops.framework import EventBase
+from adapters.framework import FrameworkAdapter
+
+
+class NewAlertManagerRelationEvent(EventBase):
+
+    def __init__(self, handle, remote_data):
+        super().__init__(handle)
+        self.data = dict(remote_data)
+
+    def snapshot(self):
+        return json.dumps(self.data)
+
+    def restore(self, snapshot):
+        self.data = json.loads(snapshot)
+
+
+class AlertManagerEvents(ObjectEvents):
+    new_am_rel = EventSource(NewAlertManagerRelationEvent)
```

Comment:
`rel` instead of `relation` again. I'd replace it with `relation`, at least from my point of view it looks better.